components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -51,8 +51,10 @@ object typing {
     def apply(fields: List[(String, TypingResult)], objType: TypedClass): TypedObjectTypingResult =
       TypedObjectTypingResult(ListMap(fields: _*), objType)
 
-    def apply(fields: ListMap[String, TypingResult]): TypedObjectTypingResult =
-      TypedObjectTypingResult(fields, TypedClass(classOf[java.util.Map[_, _]], List(Typed[String], Unknown)))
+    def apply(fields: ListMap[String, TypingResult]): TypedObjectTypingResult = {
+      val valueTypes = if (fields.isEmpty) Unknown else Typed(fields.values.toList:_*)
```

Comment:
After "online" discussion I understand better what happened here. Ignore my first comment.

Reply:
Now I put a comment that is more accurate: Take a look at the way how the `Typer` handles `InlineList`. We are looking for supertype when list is not empty. IMO we should behave the same here. I wonder if we can move this code closer to themselves to keep it consistent. WDYT? Dawid done it that way because we have a discussion that union types are rather purely supported by us. It is more safe to handle it by looking for supertype for now.

Reply:
:+1: this is v. good observation, it will be much better, thx

Reply:
Don't forgot to remove already unused variable ;)

Reply:
:+1: :)